examples/pointer.c

```diff
@@ -42,112 +88,291 @@ static void handle_output_frame(struct output_state *output, struct timespec *ts
+			// configure device to output mappings
+			if (dc->mapped_output &&
+					strcmp(dc->mapped_output, ostate->output->name) == 0) {
+				wl_list_for_each(dev, &sample->devices, link) {
+					if (strcmp(dev->device->name, dc->name) == 0) {
+						wlr_cursor_map_input_to_output(sample->cursor,
+							dev->device, ostate->output);
+					}
+				}
+			}
+		}
+	}
+}
+
+static void handle_output_add(struct output_state *ostate) {
+	struct sample_state *sample = ostate->compositor->data;
+	struct wlr_output *wlr_output = ostate->output;
+	struct wlr_xcursor_image *image = sample->xcursor->images[0];
+
+	// reset layout
+	wlr_output_layout_destroy(sample->layout);
+	sample->layout = configure_layout(sample->config, &ostate->compositor->outputs);
+	wlr_cursor_attach_output_layout(sample->cursor, sample->layout);
+
+	// cursor configuration
+	char *mapped_output = sample->config->cursor.mapped_output;
+	if (mapped_output && strcmp(mapped_output, wlr_output->name) == 0) {
+		wlr_cursor_map_to_output(sample->cursor, wlr_output);
+	}
+
+	configure_devices(sample);
+
+	// TODO move to wlr_cursor
+	if (!wlr_output_set_cursor(wlr_output, image->buffer,
+			image->width, image->width, image->height)) {
+		wlr_log(L_DEBUG, "Failed to set hardware cursor");
+		return;
+	}
+	if (!wlr_output_move_cursor(wlr_output, 0, 0)) {
+		wlr_log(L_DEBUG, "Failed to move hardware cursor");
+	}
+}
+
+static void handle_output_remove(struct output_state *ostate) {
+	struct sample_state *sample = ostate->compositor->data;
+	wlr_output_layout_destroy(sample->layout);
+	sample->layout = configure_layout(sample->config, &ostate->compositor->outputs);
+	wlr_cursor_attach_output_layout(sample->cursor, sample->layout);
+
+	configure_devices(sample);
+
+	char *mapped_output = sample->config->cursor.mapped_output;
+	if (mapped_output && strcmp(mapped_output, ostate->output->name) == 0) {
+		wlr_cursor_map_to_output(sample->cursor, NULL);
+	}
+}
+
+static void handle_output_resolution(struct compositor_state *state,
+		struct output_state *ostate) {
+	struct sample_state *sample = ostate->compositor->data;
+	wlr_output_layout_destroy(sample->layout);
+	sample->layout = configure_layout(sample->config, &ostate->compositor->outputs);
+	wlr_cursor_attach_output_layout(sample->cursor, sample->layout);
+}
+
+static void handle_input_add(struct compositor_state *state, struct
+		wlr_input_device *device) {
+	struct sample_state *sample = state->data;
+
+	if (device->type == WLR_INPUT_DEVICE_POINTER ||
+			device->type == WLR_INPUT_DEVICE_TOUCH ||
+			device->type == WLR_INPUT_DEVICE_TABLET_TOOL) {
+		struct sample_input_device *s_device;
+		s_device = calloc(1, sizeof(struct sample_input_device));
+		s_device->device = device;
+
+		wl_list_insert(&sample->devices, &s_device->link);
+		wlr_cursor_attach_input_device(sample->cursor, device);
+		configure_devices(sample);
+	}
+}
+
+static void handle_input_remove(struct compositor_state *state, struct
```

Comment:
Don't split parameter declarations across lines, put the whole thing on the next line

Reply:
(this is not the only place this is a problem)